src/preact/base-element.js

```diff
@@ -477,3 +505,55 @@ function matchChild(element, defs) {
   }
   return null;
 }
+
+/**
+ * @param {!NodeList} nodeList
+ * @return {boolean}
+ */
+function shouldMutationForNodeListBeRerendered(nodeList) {
+  for (let i = 0; i < nodeList.length; i++) {
+    const node = nodeList[i];
+    if (node.nodeType == /* ELEMENT */ 1) {
+      // Ignore service elements, e.g. `<i-amphtml-svc>` or
+      // `<x slot="i-amphtml-svc">`.
+      if (
+        startsWith(node.tagName.toLowerCase(), IGNORE_NAME) ||
+        node.getAttribute('slot') == 'i-amphtml-svc'
+      ) {
+        continue;
+      }
+      return true;
+    }
+    if (node.nodeType == /* TEXT */ 3) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * @param {typeof PreactBaseElement} Ctor
+ * @param {!MutationRecord} record
+ * @return {boolean}
+ */
+function shouldMutationBeRerendered(Ctor, m) {
+  const {target, type} = m;
+  if (type == 'attributes') {
+    // Check if the attribute is mapped to one of the properties.
+    const props = Ctor['props'];
+    for (const name in props) {
```

Comment:
Maybe we should invert the props map? Make it amp-attr -> prop-name instead of prop-name -> amp-attr?

Reply:
Yeah. thought about that as well. I'll track it separately and see what other problems need. But I think inverting would be the right thing.